File: qqbot-module-platform-admin/src/main/java/com/mieai/qqbot/admin/onboarding/OnboardingController.java
package com.mieai.qqbot.admin.onboarding;

import jakarta.validation.Valid;
import java.util.Objects;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/system/onboarding")
public class OnboardingController {
    private final OnboardingAdministrationService service;

    public OnboardingController(OnboardingAdministrationService service) {
        this.service = Objects.requireNonNull(service, "service must not be null");
    }

    @GetMapping
    public OnboardingStatusResponse status() {
        return service.status();
    }

    @PostMapping("/database-configured")
    public OnboardingStatusResponse databaseConfigured(
            @Valid @RequestBody DatabaseConfiguredRequest request) {
        return service.databaseConfigured(request.expectedRevision(), request.databaseType());
    }

    @PostMapping("/complete")
    public OnboardingStatusResponse complete() {
        return service.complete();
    }
}

File: qqbot-module-platform-admin/src/main/kotlin/com/mieai/qqbot/admin/onboarding/OnboardingController.kt
package com.mieai.qqbot.admin.onboarding

import jakarta.validation.Valid
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/system/onboarding")
class OnboardingController(
    private val service: OnboardingAdministrationService,
) {
    @GetMapping
    fun status(): OnboardingStatusResponse = service.status()

    @PostMapping("/database-configured")
    fun databaseConfigured(
        @Valid @RequestBody request: DatabaseConfiguredRequest,
    ): OnboardingStatusResponse = service.databaseConfigured(
        request.expectedRevision,
        requireNotNull(request.databaseType),
    )

    @PostMapping("/complete")
    fun complete(): OnboardingStatusResponse = service.complete()
}
